yb-voyager/cmd/importDataFileTaskImporter.go

```diff
@@ -0,0 +1,252 @@
+	if fti.batchProducer.Done() {
+		return fmt.Errorf("no more batches to submit")
+	}
+	batch, err := fti.batchProducer.NextBatch()
+	if err != nil {
+		return fmt.Errorf("getting next batch: %w", err)
+	}
+	return fti.submitBatch(batch)
+}
+
+func (fti *FileTaskImporter) importBatch(batch *Batch) {
+	err := batch.MarkPending()
+	if err != nil {
+		utils.ErrExit("marking batch as pending: %d: %s", batch.Number, err)
+	}
+	log.Infof("Importing %q", batch.FilePath)
+
+	importBatchArgs := *fti.importBatchArgsProto
+	importBatchArgs.FilePath = batch.FilePath
+	importBatchArgs.RowsPerTransaction = batch.OffsetEnd - batch.OffsetStart
+
+	var rowsAffected int64
+	sleepIntervalSec := 0
+	for attempt := 0; attempt < COPY_MAX_RETRY_COUNT; attempt++ {
+		tableSchema, _ := TableNameToSchema.Get(batch.TableNameTup)
+		rowsAffected, err = tdb.ImportBatch(batch, &importBatchArgs, exportDir, tableSchema)
+		if err == nil || tdb.IsNonRetryableCopyError(err) {
+			break
+		}
+		log.Warnf("COPY FROM file %q: %s", batch.FilePath, err)
+		sleepIntervalSec += 10
+		if sleepIntervalSec > MAX_SLEEP_SECOND {
+			sleepIntervalSec = MAX_SLEEP_SECOND
+		}
+		log.Infof("sleep for %d seconds before retrying the file %s (attempt %d)",
+			sleepIntervalSec, batch.FilePath, attempt)
+		time.Sleep(time.Duration(sleepIntervalSec) * time.Second)
+	}
+	log.Infof("%q => %d rows affected", batch.FilePath, rowsAffected)
+	if err != nil {
+		utils.ErrExit("import batch: %q into %s: %s", batch.FilePath, batch.TableNameTup, err)
+	}
+	err = batch.MarkDone()
+	if err != nil {
+		utils.ErrExit("marking batch as done: %q: %s", batch.FilePath, err)
+	}
+}
+
+func (fti *FileTaskImporter) submitBatch(batch *Batch) error {
+	fti.workerPool.Go(func() {
+		// There are `poolSize` number of competing go-routines trying to invoke COPY.
+		// But the `connPool` will allow only `parallelism` number of connections to be
+		// used at a time. Thus limiting the number of concurrent COPYs to `parallelism`.
+		fti.importBatch(batch)
+		if reportProgressInBytes {
+			fti.updateProgress(batch.ByteCount)
+		} else {
+			fti.updateProgress(batch.RecordCount)
+		}
+	})
+	log.Infof("Queued batch: %s", spew.Sdump(batch))
+	return nil
+}
+
+func (fti *FileTaskImporter) updateProgress(progressAmount int64) {
+	fti.currentProgressAmount += progressAmount
+	fti.progressReporter.AddProgressAmount(fti.task, progressAmount)
+
+	// The metrics are sent after evry 5 secs in implementation of UpdateImportedRowCount
+	if fti.totalProgressAmount > fti.currentProgressAmount {
+		fti.updateProgressInControlPlane(ROW_UPDATE_STATUS_IN_PROGRESS)
+	}
+}
+
+func (fti *FileTaskImporter) PostProcess() {
+	if fti.batchProducer != nil {
+		fti.batchProducer.Close()
+	}
+
+	fti.updateProgressInControlPlane(ROW_UPDATE_STATUS_COMPLETED)
+
+	fti.progressReporter.FileImportDone(fti.task) // Remove the progress-bar for the file.\
+}
+
+func (fti *FileTaskImporter) updateProgressInControlPlane(status int) {
```

Comment:
Please also test the yugabyteD UI once for the import-data progress if its working as expected.

Reply:
working as expected 👍  
<img width="1453" alt="image" src="https://github.com/user-attachments/assets/ed907210-97e3-4934-bedc-3367a9fe0914" />